src/main.cpp

```diff
@@ -4272,6 +4272,15 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
+    if ((pfrom->nServices & NODE_BLOOM) &&
```

Comment:
Whoops, yes! The test I mentioned in "What's missing" would have caught that. So maybe we should block this PR on writing that test.